sympy/solvers/ode/tests/test_ode.py

```diff
@@ -3701,74 +3697,69 @@ def test_sysode_linear_neq_order1():
+    sol6 = [Eq(x(t), -C1*sin(2*t) - C2*cos(2*t)), Eq(y(t), C1*cos(2*t) - C2*sin(2*t))]
     assert dsolve(eq6) == sol6
     assert checksysodesol(eq6, sol6) == (True, [0, 0])
 
     eq7 = [Eq(x(t).diff(t), I*y(t)), Eq(y(t).diff(t), I*x(t))]
     #sol7 = [Eq(x(t), C1*exp(I*t) + C2*exp(-I*t)), Eq(y(t), C1*exp(I*t) - C2*exp(-I*t))]
-    sol7 = [Eq(x(t), I*C1*exp(I*t) + I*C2*exp(-I*t)), Eq(y(t), I*C1*exp(I*t) - I*C2*exp(-I*t))]
+    sol7 = [Eq(x(t), -C1*exp(-I*t) + C2*exp(I*t)), Eq(y(t), C1*exp(-I*t) + C2*exp(I*t))]
     assert dsolve(eq7) == sol7
     assert checksysodesol(eq7, sol7) == (True, [0, 0])
 
     eq8 = [Eq(x(t).diff(t), -a*y(t)), Eq(y(t).diff(t), a*x(t))]
-    sol8 = [Eq(x(t), C1*cos(a*t) + C2*sin(a*t)), Eq(y(t), C1*sin(a*t) - C2*cos(a*t))]
+    sol8 = [Eq(x(t), -I*C1*exp(-I*a*t) + I*C2*exp(I*a*t)), Eq(y(t), C1*exp(-I*a*t) + C2*exp(I*a*t))]
     #assert dsolve(eq8) == sol8
     assert checksysodesol(eq8, sol8) == (True, [0, 0])
 
     eq9 = [Eq(x(t).diff(t), x(t) + y(t)), Eq(y(t).diff(t), x(t) - y(t))]
-    sol9 = [Eq(x(t), C1*exp(sqrt(2)*t) + C2*exp(-sqrt(2)*t)),
-            Eq(y(t), C1*(-1 + sqrt(2))*exp(sqrt(2)*t) + C2*(-sqrt(2) - 1)*exp(-sqrt(2)*t))]
+    sol9 = [Eq(x(t), C1*(1 - sqrt(2))*exp(-sqrt(2)*t) + C2*(1 + sqrt(2))*exp(sqrt(2)*t)),
+            Eq(y(t), C1*exp(-sqrt(2)*t) + C2*exp(sqrt(2)*t))]
     assert dsolve(eq9) == sol9
     assert checksysodesol(eq9, sol9) == (True, [0, 0])
 
     eq10 = [Eq(x(t).diff(t), x(t) + y(t)), Eq(y(t).diff(t), x(t) + y(t))]
-    sol10 = [Eq(x(t), C1*exp(2*t) + C2), Eq(y(t), C1 * exp(2*t) - C2)]
+    sol10 = [Eq(x(t), -C1 + C2*exp(2*t)), Eq(y(t), C1 + C2*exp(2*t))]
     assert dsolve(eq10) == sol10
     assert checksysodesol(eq10, sol10) == (True, [0, 0])
 
     eq11 = [Eq(x(t).diff(t), 2*x(t) + y(t)), Eq(y(t).diff(t), -x(t) + 2*y(t))]
-    sol11 = [Eq(x(t), (C1*cos(t) + C2*sin(t)) * exp(2*t)),
-             Eq(y(t), (-C1*sin(t) + C2*cos(t)) * exp(2*t))]
+    sol11 = [Eq(x(t), C1*exp(2*t)*sin(t) + C2*exp(2*t)*cos(t)),
+            Eq(y(t), C1*exp(2*t)*cos(t) - C2*exp(2*t)*sin(t))]
     assert dsolve(eq11) == sol11
     assert checksysodesol(eq11, sol11) == (True, [0, 0])
 
     eq12 = [Eq(x(t).diff(t), x(t) + 2*y(t)), Eq(y(t).diff(t), 2*x(t) + y(t))]
-    #sol12 = [Eq(x(t), C1*exp(-t) + C2*exp(3*t)),
-    #         Eq(y(t), -C1*exp(-t) + C2*exp(3*t))]
-    sol12 = [Eq(x(t), 2*C2*exp(-t) + 2*C1*exp(3*t)),
-             Eq(y(t), -2*C2*exp(-t) + 2*C1*exp(3*t))]
+    sol12 = [Eq(x(t), -C1*exp(-t) + C2*exp(3*t)), Eq(y(t), C1*exp(-t) + C2*exp(3*t))]
     assert dsolve(eq12) == sol12
     assert checksysodesol(eq12, sol12) == (True, [0, 0])
 
     eq13 = [Eq(x(t).diff(t), 4*x(t) + y(t)), Eq(y(t).diff(t), -x(t) + 2*y(t))]
-    sol13 = [Eq(x(t), (C1 + C2*(t+1)) * exp(3*t)),
-             Eq(y(t), -(C1 + C2*t) * exp(3*t))]
+    sol13 = [Eq(x(t), C1*exp(3*t) + C2*(t*exp(3*t) + exp(3*t))),
+            Eq(y(t), -C1*exp(3*t) - C2*t*exp(3*t))]
```

Comment:
Maybe it would be good to collect on `exp(3*t)`:
```
In [7]: Eq(x(t), C1*exp(3*t) + C2*(t*exp(3*t) + exp(3*t))).rhs                                                                    
Out[7]: 
    3⋅t      ⎛   3⋅t    3⋅t⎞
C₁⋅ℯ    + C₂⋅⎝t⋅ℯ    + ℯ   ⎠

In [8]: Eq(x(t), C1*exp(3*t) + C2*(t*exp(3*t) + exp(3*t))).rhs.collect(exp(3*t))                                                  
Out[8]: 
                   3⋅t
(C₁ + C₂⋅(t + 1))⋅ℯ  
```

Reply:
Should this collection be done only for this test case or its a post processing step that you are suggesting where we collect on terms

Reply:
I was thinking that it should happen every time. We always know that there will be particular exponential factors in the results returned from this method and we know what they are so we can collect on them.

Reply:
So first we would need to get those exponential factors known and then we can apply ```collect``` on all the solutions.